.github/workflows/main.yaml

```diff
@@ -67,40 +67,31 @@ jobs:
           key: venv-${{ runner.os }}-${{ hashFiles('**/poetry.lock') }}
 
       #----------------------------------------------
-      # install dependencies if cache does not exist 
+      # install dependencies if cache does not exist
       #----------------------------------------------
       - name: Install dependencies
         if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
         run: poetry install --no-interaction --no-root
 
       #----------------------------------------------
-      #    install your root project, if required 
-      #----------------------------------------------      
+      #    install your root project, if required
+      #----------------------------------------------
       - name: Install library
         run: poetry install --no-interaction
 
       #----------------------------------------------
-      #              run test suite   
+      #    run tests and generate coverage report
       #----------------------------------------------
       - name: Run tests
-        run: poetry run python -m unittest discover
-
-      #----------------------------------------------
-      #              coverage report   
-      #----------------------------------------------
-      - name: Generate coverage results
-        run: |
-          poetry run coverage run -m unittest discover
-          poetry run coverage xml
-          poetry run coverage report -m
+        run: poetry run pytest --cov=src/linkml_transformer --cov-report=xml -rP tests/
```

Comment:
this runs the tests using pytest and generates the code coverage at the same time. Amazing!

Reply:
@hrshdhgd @sierra-moxon let's use this pattern everywhere